modules/AOS_PDF_Templates/templateParser.php

```diff
@@ -110,11 +113,29 @@ public static function parse_template_bean($string, $key, &$focus)
                         $link = $secureLink;
                         $repl_arr[$key . "_" . $fieldName] = '<img src="' . $link . '" width="' . $field_def['width'] . '" height="' . $field_def['height'] . '"/>';
                     }
-                } elseif ($field_def['type'] == 'wysiwyg') {
+                } elseif ($fieldType === 'wysiwyg') {
                     $repl_arr[$key . "_" . $field_def['name']] = html_entity_decode($focus->$field_def['name'],
                         ENT_COMPAT, 'UTF-8');
                     $repl_arr[$key . "_" . $fieldName] = html_entity_decode($focus->{$fieldName},
                         ENT_COMPAT, 'UTF-8');
+                } elseif ($fieldType === 'datetime' || $fieldDbType === 'datetime') {
+
+                    $datetimeValue = $focus->{$fieldName} ?? '';
+                    if ($datetimeValue !== '' && is_string($datetimeValue)){
+                        $datetimeValue = self::getDateFormatService()->toUserDateTime($datetimeValue);
+                    }
+
+                    $repl_arr[$key . "_" . $fieldName] = $datetimeValue;
+
+                } elseif ($fieldType === 'date' || $fieldDbType === 'date') {
+
+                    $dateValue = $focus->{$fieldName} ?? '';
+                    if ($dateValue !== '' && is_string($dateValue)){
+                        $dateValue = self::getDateFormatService()->toUserDate($dateValue);
```

Comment:
we probably need to take account of the system format or a specific template format as opposed to the users format, it may not make sense